common/djangoapps/enrollment/views.py

```diff
@@ -489,3 +490,16 @@ def post(self, request):
                     ).format(username=username, course_id=course_id)
                 }
             )
+        finally:
+            # Assumes that the ecommerce service uses an API key to authenticate.
+            if has_api_key_permissions:
+                current_enrollment = api.get_enrollment(username, unicode(course_id))
```

Comment:
This could return `None`.  Also, the same call is made above and stored in the `enrollment` variable - i suggest initializing that outside of the try block, and handling if it's `None` in the audit log call.